packages/block-library/src/image/style.scss

```diff
@@ -11,6 +11,30 @@
 		box-sizing: border-box;
 	}
 
+	@media (prefers-reduced-motion: no-preference) {
+		&.hide-content {
+			img {
+				visibility: hidden;
+			}
+			figcaption {
+				.wp-block-gallery & {
+					visibility: hidden;
+				}
+			}
+		}
+
+		&.show-content {
+			img {
+				animation: show-content-image 0.4s;
+			}
+			figcaption {
+				.wp-block-gallery & {
+					animation: show-content-image 0.4s;
+				}
+			}
+		}
+	}
+
```

Comment:
I'm unsure if we should put the styles to hide/show the caption here or in the gallery styles. My inclination is to put them here because we need to animate the caption with the same timing as the image. What do you think?

Note that we need to restrict these caption styles to the gallery to prevent hiding captions in other circumstances when they don't overlay the image.

Reply:
What is the desired behavior with captions here? Should they be handled different for gallery vs normal images?

Right now this is **how galleries are working in this pull request**:

https://github.com/user-attachments/assets/4f9032e6-e998-48b2-ad25-1dd86a7831e1

And this is **how normal images work**:

https://github.com/user-attachments/assets/529a8f91-ce15-40ee-a4d2-b0b9feed4e5a


This is **how they would work if we hide the caption as in the galleries**:

https://github.com/user-attachments/assets/e0874bd7-4904-4bf0-bdf3-2408fefa495a



So, what is the expected UX? Should we even add a "fade-in" effect for the caption as well?

Reply:
It is a separate issue to handle the captions correctly. I'm also wondering what the rationale was for omitting the caption in the expanded image. There are reports that the caption should be included for legal reasons on some websites https://github.com/WordPress/gutenberg/issues/60469#issuecomment-2045759484. So that at least should be an option to configure for the lightbox.

Reply:
> So, what is the expected UX? Should we even add a "fade-in" effect for the caption as well?

Ok got it, so we need to account for the caption in galleries in two scenarios:
1. When the caption overlays the image
2. When the caption *does not* overlay the image

I only took into account the first issue, not the second.

In the first scenario, like in the latest fix, I think we could make the caption disappear immediately when the lightbox opens, then make it reappear at the proper time when the lightbox is closed. We could also introduce a fade and sequence it properly with the zooming animation to make it more elegant.

For the second scenario, the caption should fade out and fade in smoothly like the rest of the content outside of the image.

> I'm also wondering what the rationale was for omitting the caption in the expanded image. There are reports that the caption should be included for legal reasons on some websites https://github.com/WordPress/gutenberg/issues/60469#issuecomment-2045759484. So that at least should be an option to configure for the lightbox.

I believe there was conversation with design that the experience of a single image lightbox doesn't require a caption because one already has the context to understand what's being displayed. This is in contrast to a lightbox gallery, in which you don't have as much context.

> It is a separate issue to handle the captions correctly. 

Ok, I'll revert the most recent change, merge, and open a separate issue 👍